Base/src/main/java/io/deephaven/base/reference/PooledObjectReference.java

```diff
@@ -105,30 +106,78 @@ protected PooledObjectReference(@NotNull final REFERENT_TYPE referent) {
     }
 
     /**
-     * Get the referent. It is an error to call this method if the caller does not have any outstanding permits.
+     * Get the referent. It is an error to call this method if the caller does not have any outstanding permits. Callers
+     * are encouraged to use this in the try block of a try-finally pattern:
+     *
+     * <pre>
+     * if (!ref.acquire()) {
+     *     return;
+     * }
+     * try {
+     *     doSomethingWith(ref.get());
+     * } finally {
+     *     ref.release();
+     * }
+     * </pre>
      *
      * @return The referent if this reference has not been cleared, null otherwise (which implies an error by the
      *         caller)
      */
     @Override
+    @Nullable
```

Comment:
Technically this is `Nullable`, but annotating it as such just encourages IntelliJ to yell at us when we're using the tool correctly and are guaranteed to not get a `null` result.